File: JavaScript/sql_injection.js
const express = require('express');
const mysql = require('mysql');
const app = express();

const connection = mysql.createConnection({
    host: 'localhost',
    user: 'root',
    password: 'password',
    database: 'testdb'
});

app.get('/user', (req, res) => {
    const username = req.query.username;
    
    // SQL Injection vulnerability - user input directly concatenated
    const query = `SELECT * FROM users WHERE username = '${username}'`;
    
    connection.query(query, (error, results) => {
        if (error) {
            res.status(500).send('Database error');
            return;
        }
        res.json(results);
    });
});

app.listen(3000, () => {
    console.log('Server running on port 3000');
});

File: Java/SQLInjection.java
import java.sql.*;

public class SQLInjection {
    public static void main(String[] args) {
        if (args.length < 1) {
            System.out.println("Usage: java SQLInjection <username>");
            return;
        }
        
        String username = args[0];
        
        try {
            Connection conn = DriverManager.getConnection("jdbc:mysql://localhost:3306/testdb", "user", "password");
            Statement stmt = conn.createStatement();
            
            // SQL Injection vulnerability - user input directly concatenated
            String query = "SELECT * FROM users WHERE username = '" + username + "'";
            ResultSet rs = stmt.executeQuery(query);
            
            while (rs.next()) {
                System.out.println("User: " + rs.getString("username"));
            }
            
            rs.close();
            stmt.close();
            conn.close();
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
}
